clar2wasm/src/lib.rs

```diff
@@ -93,7 +93,18 @@ pub fn compile(
         }
     };
 
-    let generator = WasmGenerator::new(contract_analysis.clone());
+    let generator = match WasmGenerator::new(contract_analysis.clone()) {
+        Ok(g) => g,
+        Err(e) => {
+            diagnostics.push(Diagnostic::err(&e));
+            return Err(CompileError::Generic {
+                ast,
+                diagnostics,
+                cost_tracker: Box::new(contract_analysis.cost_track.take().unwrap()),
+            });
+        }
+    };
+
```

Comment:
Since we don't do anything special with the `Ok` and the `Err` is the same as the next match, I would remove this part and replace the next part with
```rust
    match WasmGenerator::new(contract_analysis.clone()).and_then(WasmGenerator::generate) {
        Ok(module) => Ok(CompileResult {
            ast,
            diagnostics,
            module,
            contract_analysis,
        }),
        Err(e) => {
            diagnostics.push(Diagnostic::err(&e));
            Err(CompileError::Generic {
                ast,
                diagnostics,
                cost_tracker: Box::new(contract_analysis.cost_track.take().unwrap()),
            })
        }
    }
```

Reply:
That's a good one. Refactored!